fuse_core/include/fuse_core/async_publisher.hpp

```diff
@@ -81,10 +82,25 @@ class AsyncPublisher : public Publisher
    * AsyncPublisher::onInit() method will be called from here, once the internal node is properly
    * configured.
    *
+   * @param[in] interfaces The node interfaces to be used with the publisher
    * @param[in] name A unique name to give this plugin instance
+   * @param[in] name The number of threads to use with this publisher
    * @throws runtime_error if already initialized
    */
-  void initialize(const std::string & name) override;
```

Comment:
I think the non-virtual `initialize()` is a problem. The optimizer [is going to call initialize(...)](https://github.com/locusrobotics/fuse/blob/6ee89e8d6b24b282fb99c14482fe2d6412dcc3e9/fuse_optimizers/src/optimizer.cpp#L343) without knowing the type of the class, so all classes need to accept the same arguments.

Maybe we could make `initialize()` accept all known interfaces, and then plugins would use only the ones they needed? I'm playing with this locally now.

Reply:
It seems that making initialize accept all known interfaces is the solution.. I was trying to avoid that so as to have the base class have as few concerns as possible, but I can't think of another way around it if the signatures have to stay the same...

Reply:
I updated https://github.com/methylDragon/fuse/pull/3 to make `Publisher::interfaces()` accept all node interfaces

Reply:
:bow: Let me take an iterative pass